File: CP2K_functions_Py/Functions_pwd/Py_Functions_files/getABCvectors.py
import numpy as np
import os
import warnings
    
def getABCvectors(ParentDir = None,WorkFldr = None): 
    if os.path.exist(str(np.array([ParentDir,WorkFldr,'\',WorkFldr,'-1.restart']))):
        fid = open(np.array([ParentDir,WorkFldr,'\',WorkFldr,'-1.restart']))
    else:
        if os.path.exist(str(np.array([ParentDir,WorkFldr,'\',WorkFldr,'.inp']))):
            fid = open(np.array([ParentDir,WorkFldr,'\',WorkFldr,'.inp']))
        else:
            if os.path.exist(str(np.array([ParentDir,WorkFldr,'\',WorkFldr,'.pdb']))):
                fid = open(np.array([ParentDir,WorkFldr,'\',WorkFldr,'.pdb']))
                lines = textscan(fid,'%s','delimiter','\n','whitespace','')
                fid.close()
                lines = lines[0]
                VectorLine = str2num(lines[0](np.arange(7,end() - 15+1)))
                Vectors = VectorLine(np.arange(1,3+1))
                return Vectors
            else:
                warnings.warn('No input or restart file detected...')
                return Vectors
    
    lines = textscan(fid,'%s','delimiter','\n','whitespace','')
    fid.close()
    lines = lines[0]
    relevant = find(not cellfun(isempty,strfind(lines,'ABC')) )
    if not len(relevant)==0 :
        VectorLine = lines[relevant]
        vecIndx = find(VectorLine == 'C')
        Vectors = str2num(VectorLine(np.arange(vecIndx + 1,end()+1)))
        return Vectors
    else:
        ALineIndx = find(not cellfun(isempty,strfind(lines,' A ')) )
        BLineIndx = find(not cellfun(isempty,strfind(lines,' B ')) )
        CLineIndx = find(not cellfun(isempty,strfind(lines,' C ')) )
        AVectorLine = lines[ALineIndx]
        vecStartA = find(AVectorLine == 'A')
        A = str2num(AVectorLine(np.arange(vecStartA + 1,end()+1)))
        BVectorLine = lines[BLineIndx]
        vecStartB = find(BVectorLine == 'B')
        B = str2num(BVectorLine(np.arange(vecStartB + 1,end()+1)))
        CVectorLine = lines[CLineIndx]
        vecStartC = find(CVectorLine == 'C')
        C = str2num(CVectorLine(np.arange(vecStartC + 1,end()+1)))
        if norm(A) == np.logical_and(sum(A),norm(B)) == np.logical_and(sum(B),norm(C)) == sum(C):
            Vectors = np.transpose(diag(np.array([[A],[B],[C]])))
        else:
            warnings.warn('Lattice vectors are not orthogonal...')
        return Vectors
    
    return Vectors

File: CP2K_functions_Py/Functions_pwd/Original_m_files/getABCvectors.m
function [Vectors] = getABCvectors(ParentDir, WorkFldr)

if exist([ParentDir WorkFldr '\' WorkFldr '-1.restart'], 'file')
    fid  = fopen([ParentDir WorkFldr '\' WorkFldr '-1.restart']);
elseif exist([ParentDir WorkFldr '\' WorkFldr '.inp'], 'file')
    fid  = fopen([ParentDir WorkFldr '\' WorkFldr '.inp']);
elseif exist([ParentDir WorkFldr '\' WorkFldr '.pdb'], 'file')
    fid  = fopen([ParentDir WorkFldr '\' WorkFldr '.pdb']);
    lines = textscan(fid,'%s','delimiter','\n', 'whitespace', '');
    fclose(fid);
    lines = lines{1};
    VectorLine = str2num(lines{1}(7:end-15));
    Vectors = VectorLine(1:3);
    return
else
    warning('No input or restart file detected...')
    return
end

lines = textscan(fid,'%s','delimiter','\n', 'whitespace', '');
fclose(fid);
lines = lines{1};
relevant =  find(~cellfun(@isempty,strfind(lines,'ABC')));

if ~isempty(relevant)
    VectorLine = lines{relevant};
    vecIndx = find(VectorLine == 'C');
    Vectors = str2num(VectorLine(vecIndx+1:end));
    return
else
    ALineIndx = find(~cellfun(@isempty,strfind(lines,' A ')));
    BLineIndx = find(~cellfun(@isempty,strfind(lines,' B ')));
    CLineIndx = find(~cellfun(@isempty,strfind(lines,' C ')));
    
    
    AVectorLine = lines{ALineIndx};
    vecStartA = find(AVectorLine == 'A');
    A = str2num(AVectorLine(vecStartA+1:end));
    
    
    BVectorLine = lines{BLineIndx};
    vecStartB = find(BVectorLine == 'B');
    B = str2num(BVectorLine(vecStartB+1:end));
    
    CVectorLine = lines{CLineIndx};
    vecStartC = find(CVectorLine == 'C');
    C = str2num(CVectorLine(vecStartC+1:end));
    
    if norm(A) == sum(A) & norm(B) == sum(B) & norm(C) == sum(C)
        Vectors = diag([A; B; C])';
    else
        warning('Lattice vectors are not orthogonal...')
    end
    return
end